csrc/kernel_cache.cpp

```diff
@@ -283,8 +298,8 @@ KernelArgumentHolder FusionExecutorCache::prepareInputs(
   // fusions. This may not be ideal in all cases, since it will prevent
   // short-circuiting here, resulting in avoidable rebuilds of concretization
   // info.
-  auto id_lookup_ret =
-      inputs_id_lookup_.lookupId(inputs, /*hash_scalars*/ isDynamic());
+  auto id_lookup_ret = inputs_id_lookup_.lookupId(
+      inputs, &(initialInfo().inputsAffectConcretization()));
```

Comment:
Not quite sure if this is safe. Looks like a pointer to a temporary is passed to `lookupId`. Isn't the temporary invalid when `lookupId` is executed? Or, does C++ guarantee the lifetime of the temporary is extended like its const reference is passed?

Either way, just passing its const reference seems simpler and better.

Reply:
The const reference is indeed better if the argument is not optional. I used a pointer instead to handle a default value, but that's actually more trouble than it's worth. I'll change it to a const reference with an override for when the argument is missing.